### PR TITLE
v10 - support react-intl v3

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Preparation for next release. Now UI-Kit can be used with the latest (beta) version of react-intl. 

This PR is aimed at v10

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,11 +53,11 @@
     "format:js": "prettier --write '**/*.js'",
     "format:md": "prettier --write --parser markdown '**/*.md'",
     "format:yaml": "prettier --write --parser yaml '**/*.yaml'",
-    "test": "jest --config jest.test.config.js",
-    "test:watch": "jest --config jest.test.config.js --watch",
-    "test:bundle": "jest --config jest.bundle.config.js",
+    "test": "cross-env NODE_ICU_DATA=node_modules/full-icu jest --config jest.test.config.js",
```
Comment (montezume) [inline package.json]: From here

https://github.com/formatjs/react-intl/issues/1179#issuecomment-428183019

Comment (montezume) [reply package.json]: Otherwise tests that rely on locale stuff will fail

File: package.json [hunk 2]
```diff
@@ -53,11 +53,11 @@
...
+    "test:watch": "cross-env NODE_ICU_DATA=node_modules/full-icu jest --config jest.test.config.js --watch",
+    "test:bundle": "cross-env cross-env NODE_ICU_DATA=node_modules/full-icu jest --config jest.bundle.config.js",
     "bundlesize": "bundlesize",
-    "test:bundle:watch": "jest --config jest.bundle.config.js --watch",
+    "test:bundle:watch": "cross-env NODE_ICU_DATA=node_modules/full-icu jest --config jest.bundle.config.js --watch",
```
Comment (montezume) [inline package.json]: maybe don't need this for these

File: package.json (hunk 1 above)
Comment (tdeekens) [inline package.json]: Sadly npm hooks don't support wildcards (e.g. `pretest*`) but maybe a `setup` run script is nice to ease this and then just have `yarn setup && jest ...`.

Comment (montezume) [reply package.json]: Hm... can you clarify what `setup` would look like? 😄

Comment (emmenko) [reply package.json]: In appkit I did it like this:

https://github.com/commercetools/merchant-center-application-kit/blob/b6200086597c97640852052618cedc7cdb17f3b5/packages/jest-preset-mc-app/setup-test-framework.js#L26-L31

File: package.json [hunk 3]
```diff
@@ -206,7 +207,7 @@
     "moment-timezone": "0.5.x",
     "react": "16.x",
     "react-dom": "16.x",
-    "react-intl": "2.x",
+    "react-intl": "2.x || 3.x",
```
Comment (emmenko) [inline package.json]: Are you sure we can make it backwards compatible? Can't we just include it in v10?

File: package.json (hunk 1 above)
Comment (tdeekens) [reply package.json]: `"setup": "cross-env NODE_ICU_DATA=node_modules/full-icu"` :D

Comment (montezume) [reply package.json]: I tried this first 😢 

I think it's a different issue.

File: package.json (hunk 3 above)
Comment (montezume) [reply package.json]: I think the only real breaking change is the dropping of `intlShape` and the polyfills.

Comment (montezume) [reply package.json]: Maybe to err on the safe side we should aim this at v10 though...

File: package.json (hunk 1 above)
Comment (montezume) [reply package.json]: I tried this and it didn't work 🤔 ? In the test I put yarn setup && jest ...

Comment (emmenko) [reply package.json]: I don't think installing the full icu is necessary. Patching the Intl in the jest setup fixed the problem for me.

What is the problem you're having?

Comment (tdeekens) [reply package.json]: Ah spawned in different shell maybe.

Comment (montezume) [reply package.json]: Turns out just needed more polyfills 😓

**Comment by montezume** [comment]
Alright this is ready for review 😇

**Comment by montezume** [comment]
Does nobody want to review this 🤔

**Pull request merged**